programs/nft-club/src/components/benefit.rs

```diff
@@ -36,6 +36,22 @@ pub fn delete_benefit(ctx: Context<DeleteBenefit>,  _benefit_number: String) ->
     Ok(())
 }
 
+pub fn update_benefit(
+    ctx: Context<UpdateBenefit>, 
+    name: String, 
+    description: String, 
+    _benefit_number: String
+) -> Result<()> {
+    let benefit: &mut Account<Benefit> = &mut ctx.accounts.benefit;
+    let authority: &Signer = &ctx.accounts.authority;
+
+    benefit.authority = *authority.key;
```

Comment:
We likely shouldn't ever have a scenario where we update the `authority` right? Should probably remove this

Reply:
Very good point. Thanks for catching this and pointing it out!